### ISSUE TITLE
Automatic configuration won't start in Angular 13.

### ISSUE THREAD

**Issue by seiryokusan** [wallabyjs/public#2856]
I started a new project with Angular 13, and the automatic configuration throws this error:
```
‌[Error] Automatic Angular CLI configuration error: x is not a function 
[Error]  
[Error] We've also tried to automatically configure Wallaby.js for other frameworks 
[Error] Automatic Jest configuration error: Module jest-cli is not found in '.'. 
```
I had node 16.13.0, but I figured I would downgrade to 14.18.1 in case my node was too new, but it did not help.
I also installed jest-cli in case it would help, but it did not.
`ng test` work as intended.

### Wallaby diagnostics report
```
{
  editorVersion: '1.62.3',
  pluginVersion: '1.0.318',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.19042',
  nodeVersion: 'v14.18.1',
  coreVersion: '1.0.1029',
  checksum: 'NDlmYTFlODAxNTc4ODk0MTI2YTAxYmJlMTc2NWQ2ZmEsMTYzODQ4OTYwMDAwMCww',
  config: {},
  packageJSON: {
    dependencies: {
      '@angular/animations': '~13.0.0',
      '@angular/common': '~13.0.0',
      '@angular/compiler': '~13.0.0',
      '@angular/core': '~13.0.0',
      '@angular/forms': '~13.0.0',
      '@angular/platform-browser': '~13.0.0',
      '@angular/platform-browser-dynamic': '~13.0.0',
      '@angular/router': '~13.0.0',
      'pokenode-ts': '^1.9.1',
      rxjs: '~7.4.0',
      tslib: '^2.3.0',
      'zone.js': '~0.11.4'
    },
    devDependencies: {
      '@angular-devkit/build-angular': '~13.0.3',
      '@angular/cli': '~13.0.3',
      '@angular/compiler-cli': '~13.0.0',
      '@types/jasmine': '~3.10.0',
      '@types/node': '^12.11.1',
      'jasmine-core': '~3.10.0',
      karma: '~6.3.0',
      'karma-chrome-launcher': '~3.1.0',
      'karma-coverage': '~2.0.3',
      'karma-jasmine': '~4.0.0',
      'karma-jasmine-html-reporter': '~1.7.0',
      typescript: '~4.4.3'
    }
  },
  fs: { numberOfFiles: 0 },
  debug: [
    '2021-11-28T18:23:37.823Z extended-core Validating Open Source License\n',
    "2021-11-28T18:23:37.823Z extended-core Remotes:  [ 'https://github.com/seiryokusan/pokedex-petal/' ]\n",
    '2021-11-28T18:23:37.843Z angular/cli config Detected Angular CLI.\n',
    '2021-11-28T18:23:40.287Z angular/cli config TypeError: x is not a function\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:16:18387\n' +
      '    at generateWebpackConfig (G:\\WorkWork\\pokedex-petal\\node_modules\\@angular-devkit\\build-angular\\src\\utils\\webpack-browser-config.js:61:40)\n' +
      '    at async generateBrowserWebpackConfigFromContext (G:\\WorkWork\\pokedex-petal\\node_modules\\@angular-devkit\\build-angular\\src\\utils\\webpack-browser-config.js:123:20)\n',
    '2021-11-28T18:23:40.287Z angular/cli config Angular.json: \n' +
      ' {\n' +
      "  '$schema': './node_modules/@angular/cli/lib/config/schema.json',\n" +
      '  cli: { analytics: false },\n' +
      '  version: 1,\n' +
      "  newProjectRoot: 'projects',\n" +
      '  projects: {\n' +
      "    'pokedex-petal': {\n" +
      "      projectType: 'application',\n" +
      "      schematics: { '@schematics/angular:component': { style: 'scss' }, '@schematics/angular:application': { strict: true } },\n" +
      "      root: '',\n" +
      "      sourceRoot: 'src',\n" +
      "      prefix: 'app',\n" +
      '      architect: {\n' +
      '        build: {\n' +
      "          builder: '@angular-devkit/build-angular:browser',\n" +
      '          options: {\n' +
      "            outputPath: 'dist/pokedex-petal',\n" +
      "            index: 'src/index.html',\n" +
      "            main: 'src/main.ts',\n" +
      "            polyfills: 'src/polyfills.ts',\n" +
      "            tsConfig: 'tsconfig.app.json',\n" +
      "            inlineStyleLanguage: 'scss',\n" +
      "            assets: [ 'src/favicon.ico', 'src/assets' ],\n" +
      "            styles: [ 'src/styles.scss' ],\n" +
      '            scripts: []\n' +
      '          },\n' +
      '          configurations: {\n' +
      '            production: {\n' +
      "              budgets: [ { type: 'initial', maximumWarning: '500kb', maximumError: '1mb' }, { type: 'anyComponentStyle', maximumWarning: '2kb', maximumError: '4kb' } ],\n" +
      "              fileReplacements: [ { replace: 'src/environments/environment.ts', with: 'src/environments/environment.prod.ts' } ],\n" +
      "              outputHashing: 'all'\n" +
      '            },\n' +
      '            development: { buildOptimizer: false, optimization: false, vendorChunk: true, extractLicenses: false, sourceMap: true, namedChunks: true }\n' +
      '          },\n' +
      "          defaultConfiguration: 'production'\n" +
      '        },\n' +
      '        serve: {\n' +
      "          builder: '@angular-devkit/build-angular:dev-server',\n" +
      "          configurations: { production: { browserTarget: 'pokedex-petal:build:production' }, development: { browserTarget: 'pokedex-petal:build:development' } },\n" +
      "          defaultConfiguration: 'development'\n" +
      '        },\n' +
      "        'extract-i18n': { builder: '@angular-devkit/build-angular:extract-i18n', options: { browserTarget: 'pokedex-petal:build' } },\n" +
      '        test: {\n' +
      "          builder: '@angular-devkit/build-angular:karma',\n" +
      '          options: {\n' +
      "            main: 'src/test.ts',\n" +
      "            polyfills: 'src/polyfills.ts',\n" +
      "            tsConfig: 'tsconfig.spec.json',\n" +
      "            karmaConfig: 'karma.conf.js',\n" +
      "            inlineStyleLanguage: 'scss',\n" +
      "            assets: [ 'src/favicon.ico', 'src/assets' ],\n" +
      "            styles: [ 'src/styles.scss' ],\n" +
      '            scripts: []\n' +
      '          }\n' +
      '        }\n' +
      '      }\n' +
      '    }\n' +
      '  },\n' +
      "  defaultProject: 'pokedex-petal'\n" +
      '}\n',
    '2021-11-28T18:23:40.287Z angular/cli config TypeError: x is not a function\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:16:18387\n' +
      '    at generateWebpackConfig (G:\\WorkWork\\pokedex-petal\\node_modules\\@angular-devkit\\build-angular\\src\\utils\\webpack-browser-config.js:61:40)\n' +
      '    at async generateBrowserWebpackConfigFromContext (G:\\WorkWork\\pokedex-petal\\node_modules\\@angular-devkit\\build-angular\\src\\utils\\webpack-browser-config.js:123:20)\n',
    "2021-11-28T18:23:40.391Z jest/config Error: Module jest-cli is not found in 'G:\\WorkWork\\pokedex-petal'.\n" +
      '    at Object.<anonymous> (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:29:2848)\n' +
      '    at n (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:28:19397)\n' +
      '    at Object.next (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:28:18692)\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:28:18601\n' +
      '    at new Promise (<anonymous>)\n' +
      '    at r (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:28:18395)\n' +
      '    at Object.loadJest (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:29:687)\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:12163\n' +
      '    at n (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:16612)\n' +
      '    at Object.next (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15907)\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15816\n' +
      '    at new Promise (<anonymous>)\n' +
      '    at s (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15610)\n' +
      '    at n (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:11851)\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15287\n' +
      '    at n (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:16612)\n' +
      '    at Object.next (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15907)\n' +
      '    at c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15816\n' +
      '    at new Promise (<anonymous>)\n' +
      '    at s (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15610)\n' +
      '    at r (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:18:15168)\n' +
      '    at Config.<anonymous> (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:15:24102)\n' +
      '    at n (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:15:17489)\n' +
      '    at Object.next (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:15:16784)\n' +
      '    at o (c:\\Users\\jessi\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.318\\wallabybd7fe6\\server.js:15:16533)\n'
  ]
}
```

**Comment by smcenlly** [comment]
We have tested with Angular 13 and everything appears to be working for us. I can see from your diagnostics report that you are on quite an old version (Feb 2021). Could you please update Wallaby core to the latest version, it should work for you.

If your license has expired, you may [request a trial from our website](https://wallabyjs.com/download/#try-it-free), or else contact us: hello@wallabyjs.com.

If you're still having problems with the latest version, please let us know and we will re-open the issue.

**Comment by seiryokusan** [comment]
I have updated and it works!
I had the 7-day trial license in February. I have updated to an open-source license yesterday, but it did not update automatically for some reason. I had to go to the extension page and do a "Force Core Update". 

Thanks for your help!

**Comment by seiryokusan** [comment]
I have a small problem. Any call to spyOn().and.resolveTo() gives me the error `n.getSpy is not a function`. However, it works when I use `ng test`. Is there a possibility that the jasmine version would be different when running Wallaby? I can always replace them with returnValue if necessary, but I guess it should be working?

Thanks again!

**Comment by smcenlly** [comment]
> Is there a possibility that the jasmine version.

Yes - we'll check it out. Thanks for letting us know

**Comment by smcenlly** [comment]
It seems to be working for us with a new Angular 13 project:

**example.spec.ts**
```typescript
class Dictionary {
  hello() {
    return "hello";
  }

  world() {
    return "world";
  }
}

class Person {
  sayHelloWorld(dict: Dictionary) {
    return dict.hello() + " " + dict.world();
  }
}

describe("Example Of jasmine Spy using spyOn()", function() {

  it('uses the dictionary to say "hello world"', function() {
     var dictionary = new Dictionary() as any;
     var person = new Person() as any;

     spyOn(dictionary, "hello");  // replace hello function with a spy
     spyOn(dictionary, "world");  // replace world function with another spy

     person.sayHelloWorld(dictionary);
     expect(dictionary.hello).toHaveBeenCalled();
     // not possible without first spy

     expect(dictionary.world).toHaveBeenCalled();
     // not possible withoutsecond spy
  });
});
```

Can you please see if the example above works for you? If it does and your code is not working, could you please share the code that's not working so that we can try to reproduce the problem?

**Comment by seiryokusan** [comment]
I got the code in my project.
This works: 
![image](https://user-images.githubusercontent.com/21677362/143903026-9641d2ce-31c2-48df-bd71-7a4705184b68.png)
This does not work:
![image](https://user-images.githubusercontent.com/21677362/143903164-1e70eb7b-8f25-4a3c-8b4f-5b067bdf6e58.png)

Here is my project if you want to try it with my code: https://github.com/seiryokusan/pokedex-petal
Thanks!

**Comment by smcenlly** [comment]
Thanks for the sample code. Wallaby had not actually supported `resolveTo` or `rejectWith` since they were introduced in Jasmine 3.5. We have added support for this (which will fix your problem) in Wallaby core v1.0.1180.

Wallaby should update automatically for you, but if it does not, you may force an update with the VS Code command: `Wallaby.js: Force Core Update`.
